### PR TITLE
Improve data docs

### PR DESCRIPTION
**Pull request by akihironitta**
### Description of the changes
- Adds `Deserializer`
- Includes minor fixes/improvements, e.g. links/formatting/...

## Before submitting
- [n/a] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [n/a] Did you make sure to update the documentation with your changes?
- [n/a] Did you write any new necessary tests? [not needed for typos/docs]
- [x] Did you verify new and existing tests pass locally with your changes?
- [n/a] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/general/data.rst [hunk 1]
```diff
@@ -455,7 +455,7 @@ Here is the pseudo-code:
 
 Example::
 
-    # This will be wrapped into a :class:`~flash.core.data.batch._PreProcessor`
+    # This will be wrapped into a :class:`~flash.core.data.batch._PostProcessor`
```
Comment (akihironitta) [inline docs/source/general/data.rst:458]: This should be `_PostProcessor`, right?

Comment (ethanwharris) [reply docs/source/general/data.rst:458]: Yes, that sounds correct :)

File: docs/source/general/data.rst [hunk 2]
```diff
@@ -21,25 +21,28 @@ Here are common terms you need to be familiar with:
 
    * - Term
      - Definition
+   * - :class:`~flash.core.data.process.Deserializer`
+     - The :class:`~flash.core.data.process.Deserializer` provides a single :meth:`~flash.core.data.process.Deslerializer.deserialize` method.
```
[the remark this answers is not in the document]
Comment (ethanwharris) [reply docs/source/general/data.rst:25]: @akihironitta The Deserializer is used for serving. It's exclusion from docs is probably just an accident haha :smiley:

**Review by ethanwharris — APPROVED** [review]
@akihironitta Great work here :smiley: shall we merge this one so you don't get out of sync in #555?

File: docs/source/general/data.rst (hunk 2 above)
Comment (akihironitta) [inline docs/source/general/data.rst:25]:
```suggestion
     - The :class:`~flash.core.data.process.Deserializer` provides a single :meth:`~flash.core.data.process.Deserializer.deserialize` method.
```

**Comment by akihironitta** [comment]
@ethanwharris Sure, sounds good!

**Pull request merged**